### ISSUE TITLE
Linux Platform

### ISSUE THREAD

**Issue by iAmG-r00t** [lambdalisue/battery.vim#10]
So I have placed this plugin on my vimrc, so I don't know if its the configuration I have placed wrongly or is it because my laptop has two batteries a lenovo thinkpad T480. It doesn't show anything?

Here are snippets of my config;

![image](https://user-images.githubusercontent.com/29776892/101555779-5f4d2380-39ca-11eb-851f-a7f7c1070616.png)

![image](https://user-images.githubusercontent.com/29776892/101555759-53616180-39ca-11eb-9ca1-85152fb8c233.png)

![image](https://user-images.githubusercontent.com/29776892/101555729-40e72800-39ca-11eb-8ce4-7e1d7e0b191e.png)

**Comment by iAmG-r00t** [comment]
So an update, I have managed to get it working, by stealing some scripts from @whisperity  from this issue: #8.

here is what I added;

![image](https://user-images.githubusercontent.com/29776892/101587954-55d9b080-39f6-11eb-8f69-c76130dd9137.png)

But now the issue is it only shows percentage of my  main battery and I would love to see both;

![image](https://user-images.githubusercontent.com/29776892/101588014-773a9c80-39f6-11eb-9767-bb2bea7e9212.png)

![image](https://user-images.githubusercontent.com/29776892/101587874-25921200-39f6-11eb-863c-159641fd2e32.png)

Then another issue is, I have the update_statusline, but it doesn't update if its on charging or battery only works if I leave vim them start it once again.

**Note:** I am a new user in vim trying to learn it, and also I switched from powerline-status to airline so could you kindly assist. Would much appreciate the help.

**Comment by whisperity** [comment]
Oh... well, you could go around enumerating every `capacity` file under `/sys/class/power_supply/BAT*` (I guess they will be `BAT0` and `BAT1`?) and adding together the two values, then normalising them by dividing with the number of batteries present. The current script only reads the "first" of the wildcard expression, i.e. `BAT0`.

In pure Bash, you'd do something like this "oneliner" for that: 

```bash
echo $(( $(cat /sys/class/power_supply/BAT*/capacity | awk '{s += $1} END {print s}') / $(cat /sys/class/power_supply/BAT*/capacity | wc -l) ))
```

This will basically print out the average value. In your case, should be `33`, which basically means you've 33% of your total capacity (normal to 100% total) left (because you have 67% of 200% in the real).

I'm sure this is easier (and more readable) in proper Vim than by using `awk`, but I have not much experience with Vim scripts either.

I've never ever seen in my life a laptop with dual batteries...

**Comment by whisperity** [comment]
Am not sure about the statusline issue. When the feature was put in, I tested it extensively with Airline, and the update of status seemed to work fine. It needs a bit of time, it does not poll the battery constantly and update automatically, but should do so every few seconds.

I worked with the Airline people to have the battery plugin updated to be in a more concise location: https://github.com/vim-airline/vim-airline/pull/2243 Not sure if your airline is up-to-date with that patch.
